### ISSUE TITLE
Thank you!

### ISSUE THREAD

**Issue by gregrae** [m1cr0lab-esp32/remote-control-with-websocket#7]
Thank you sir for an exceptional project!!
This has helped me so much in my websockets-on-esp32 discovery.
cheers,
Greg

**Comment by m1cr0lab** [comment]
Hi Greg,

It's a great reward for me to know that I was able to help you and that the tutorial has really brought you something.

Thank you for your feedback.

**Comment by gregrae** [comment]
So helpful!!
One small snag i hit - in vs/Platformio I could not find "upload filesystem image" but eventually found it under environment/Platform

I am moving forward and adding a second esp32 to act as a websocket client.  I have seen the documentation at
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_websocket_client.html#
Do you know of any other sample code I might study for the websocket client?

![uploadfs](https://user-images.githubusercontent.com/4584011/97221916-cc925580-178a-11eb-9c47-e2304aad3f93.jpg)

**Comment by m1cr0lab** [comment]
Yes, indeed. It has already been pointed out to me and you can read [my answer](https://github.com/m1cr0lab-esp32/remote-control-with-websocket/issues/6#issuecomment-716323282).

As far as programming a WebSocket client is concerned, I refer you to [Espressif's documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_websocket_client.html) on the subject.
I haven't tested this possibility myself, but I guess you can also take inspiration from [this tutorial](https://techtutorialsx.com/2017/11/01/esp32-arduino-websocket-client/).
Maybe there are more recent libraries... I don't know, you have to search a little ;-)

However, if you simply want to make two ESP32s communicate with each other, you don't need this layer. You can use the connectionless ESP-NOW protocol designed by Espressif. It's much lighter and much less energy consuming than a WiFi connection! You will find many tutorials on the subject here and there, especially on [RNT](https://randomnerdtutorials.com/esp-now-esp32-arduino-ide/). Who knows? Maybe I'll write one myself one day...

**Comment by gregrae** [comment]
Thank you Steph!  You are very kind.

**Comment by m1cr0lab** [comment]
@gregrae Hi Greg!

I found this [WebSocket Sample Application](https://github.com/espressif/esp-idf/tree/357a277603/examples/protocols/websocket) that should be useful to you ;-)
It uses Espressif's ESP WebSocket Client.

The [code](https://github.com/espressif/esp-idf/blob/357a2776032299b8bc4044900a8f1d6950d7ce89/examples/protocols/websocket/main/websocket_example.c) will give you the keys you are looking for I guess.

**Comment by gregrae** [comment]
Thank you Steph!  This is great!  If my concept goes further I could see
needing a consultant/freelancer.  Would you consider taking on contract
work?
cheers,
Greg
www.gregrae.com
